workflow/workflow_test.go

```diff
@@ -308,6 +308,73 @@ func TestWithDoubleSerializationDetection(t *testing.T) {
 	assert.Equals(t, errorDetect.SerializeCnt+errorDetect.UnserializeCnt > 0, true)
 }
 
+var simpleValidLiteralInputWaitWorkflowWithDefaults = `
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties: 
+        wait_time_def:
+          type:
+            type_id: integer
+          default: 1
+          required: false
+steps:
+  wait_1:
+    plugin:
+      src: "n/a"
+      deployment_type: "builtin"
+    step: wait
+    input:
+      wait_time_ms: !expr $.input.wait_time_def
+outputs:
+  a:
+    b: !expr $.steps.wait_1.outputs
+`
+
+func Test_DoubleSerializationDetection_Default(t *testing.T) {
+	// Just a single wait
+	preparedWorkflow := assert.NoErrorR[workflow.ExecutableWorkflow](t)(
+		getTestImplPreparedWorkflow(t, simpleValidLiteralInputWaitWorkflowWithDefaults),
+	)
+	// First, get the root object
+	inputSchema := preparedWorkflow.Input()
+	rootObject := inputSchema.Objects()[inputSchema.Root()]
+	errorDetect := util.NewInvalidSerializationDetectorSchema()
+	// Inject the error detector into the object
+	rootObject.PropertiesValue["error_detector"] = schema.NewPropertySchema(
+		errorDetect,
+		nil,
+		true,
+		nil,
+		nil,
+		nil,
+		nil,
+		nil,
+	)
+	outputID, _, err := preparedWorkflow.Execute(context.Background(), map[string]any{
+		"error_detector": "original input",
+	})
+	assert.NoError(t, err)
+	assert.Equals(t, outputID, "a")
+	// Confirm that, while we did no double-unserializations or double-serializations,
+	// we did do at least one single one.
+	assert.Equals(t, errorDetect.SerializeCnt+errorDetect.UnserializeCnt > 0, true)
+
+	// again, but override the default by supplying input
```

Comment:
If we're going to reuse the `errorDetect` object (as opposed to injecting a fresh instantiation of the schema) in the second run, we should probably reset `errorDetect.SerializeCnt` and `errorDetect.UnserializeCnt` to zero after this assertion; otherwise, the assertion at line 375 won't mean anything (that is, if the assertion at line 364 passed, there's no reason to expect that the one at 375 will fail unless something _decrements_ those two counters...).

Reply:
done